src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -46,5 +50,47 @@ def contract_update_list_success(updated_resource, resource_client):
     # should be able to use the created model
     # to read since physical resource id is immutable
     _create_request, _created_model, _update_request, updated_model = updated_resource
+    resource_client.assert_primary_identifier_not_updated(
+        resource_client.primary_identifier_paths, _created_model, updated_model
+    )
     models = test_list_success(resource_client, updated_model)
     assert updated_model in models
+
+
+@pytest.mark.update
+def contract_update_invalid_property(resource_client):
```

Comment:
Please use more explicit naming in test. It's not just an invalid property, it's a createOnlyProperties check right?

Reply:
sure, updated the name to: contract_update_create_only_property